py_zipkin/encoding/_decoders.py

```diff
@@ -220,20 +219,19 @@ def _decode_thrift_span(self, thrift_span):
             thrift_span.trace_id_high,
         )
 
-        return SpanBuilder(
+        return Span(
             trace_id=trace_id,
             name=thrift_span.name,
             parent_id=parent_id,
             span_id=self._convert_unsigned_long_to_lower_hex(thrift_span.id),
+            kind=kind,
             timestamp=self.seconds(timestamp or thrift_span.timestamp),
             duration=self.seconds(duration or thrift_span.duration),
+            local_endpoint=local_endpoint,
+            remote_endpoint=remote_endpoint,
+            shared=thrift_span.timestamp is None,
```

Comment:
this is probably incorrect or at least imprecise (checking kind is server can make it more precise unless that is implicit somewhere)

Reply:
so check both if timestamp is None and if kind=SERVER?